src/Sentry.Unity/ApplicationFocusListener.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using UnityEngine;
+
+namespace Sentry.Unity
+{
+    /// <summary>
+    ///  A MonoBehavior used to forward application focus events to subscribers.
+    /// </summary>
+    [DefaultExecutionOrder(-900)]
+    internal class ApplicationFocusListener : MonoBehaviour
+    {
+        /// <summary>
+        /// Hook to receive an event when the application gains focus.
+        /// <remarks>
+        /// Listens to OnApplicationFocus for all platforms except Android, where we listen to OnApplicationPause.
+        /// </remarks>
+        /// </summary>
+        public event Action? ApplicationFocusGaining;
+
+        /// <summary>
+        /// Hook to receive an event when the application loses focus.
+        /// <remarks>
+        /// Listens to OnApplicationFocus for all platforms except Android, where we listen to OnApplicationPause.
+        /// </remarks>
+        /// </summary>
+        public event Action? ApplicationFocusLosing;
+
+        /// <summary>
+        /// To receive Leaving/Resuming events on Android.
+        /// <remarks>
+        /// https://docs.unity3d.com/2019.4/Documentation/ScriptReference/MonoBehaviour.OnApplicationPause.html
+        /// On Android, when the on-screen keyboard is enabled, it causes a OnApplicationFocus(false) event.
+        /// Additionally, if you press "Home" at the moment the keyboard is enabled, the OnApplicationFocus() event is
+        /// not called, but OnApplicationPause() is called instead.
+        /// </remarks>
+        /// </summary>
+        private void OnApplicationPause(bool pauseStatus)
+        {
+            if (Application.platform != RuntimePlatform.Android)
+            {
+                return;
+            }
+
+            if (pauseStatus)
+            {
+                ApplicationFocusLosing?.Invoke();
+            }
+            else
+            {
+                ApplicationFocusGaining?.Invoke();
+            }
+        }
+
+        /// <summary>
+        /// To receive Leaving/Resuming events on all platforms except Android.
+        /// </summary>
+        /// <param name="hasFocus"></param>
+        private void OnApplicationFocus(bool hasFocus)
+        {
+            // To avoid event duplication on Android
```

Comment:
What does it mean? Why duplication is only relevant to Android?

Reply:
I added to the comment. According to testing and the Unity docs the OnAppliationFocus event is somewhat flaky on [Android](https://docs.unity3d.com/2019.4/Documentation/ScriptReference/MonoBehaviour.OnApplicationFocus.html#:~:text=On%20Android%2C%20when%20the%20on-screen%20keyboard%20is%20enabled%2C%20it%20causes%20an%20OnApplicationFocus(%20false%20)%20event.%20Additionally%2C%20if%20you%20press%20Home%20at%20the%20moment%20the%20keyboard%20is%20enabled%2C%20the%20OnApplicationFocus()%20event%20is%20not%20called%2C%20but%20OnApplicationPause()%20is%20called%20instead.), specifically in combination with the on-screen keyboard. It's possible to leave the app without receiving the focus event. That's why we trigger the SDK sessions via OnApplicationPause on Android instead.